tests/D2L.CodeStyle.Analyzers.Test/Async/Generator/SyncGeneratorTests.cs

```diff
@@ -0,0 +1,180 @@
+﻿using D2L.CodeStyle.Analyzers.Async.Generator;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using NUnit.Framework;
+
+namespace D2L.CodeStyle.Analyzers.Tests.Async.Generator;
+
+/// <summary>
+/// Integration tests for the generator as a whole.
+/// Prefer unit tests of AsyncToSyncTransformer and FileCollector as they are
+/// simpler. We just want a bit of smoke-testing here.
+/// </summary>
+[TestFixture]
+internal sealed class SyncGeneratorTests {
+	[Test]
+	public void EmptyAssembly() {
+		var result = RunGenerator( sources: Array.Empty<string>() );
+
+		// We shouldn't have crashed anyway
+
+		AssertNoNewTrees( result );
+	}
+
+	[Test]
+	public void OneFileNoMethods() {
+		var result = RunGenerator( @"
+using System;
+
+namespace Foo;
+
+sealed class Bar {
+	public void Baz() => Console.WriteLine( ""hello"" );
+	public int Add( int x, int y ) => x + y;
+}"
+		);
+
+		AssertNoNewTrees( result );
+	}
+
+	[Test]
+	public void OneFileWithOneMethod() {
+		var result = RunGenerator( @"
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using D2L.CodeStyle.Annotations;
+
+namespace Foo;
+
+sealed class Bar {

```

Comment:
nit: `partial`? Is this (or will this be) caught by a separate analyzer that isn't running in these tests?

```suggestion
sealed partial class Bar {
```

Reply:
Ah good eye. So it won't be -- you'll get the standard error about "this needs to be partial" in the _source_ file which is actually kinda nice.

It would be reasonable to use `partial` in the test case but its not the worst to accidentally/indirectly test this I guess. 🤷